lib/dialects/abstract/connection-manager.js

```diff
@@ -295,6 +295,11 @@ class ConnectionManager {
   releaseConnection(connection) {
     return this.pool.release(connection).tap(() => {
       debug('connection released');
+    }).catch((e) => {
+      // ISSUE(7625) No-op if generic-pool is unable to release the connection, it may have already been released.
+      if (!/Resource not currently part of this pool/.exec(e.message)) {
```

Comment:
generic-pool throws a generic error which can be seen here; https://github.com/coopernurse/node-pool/blob/master/lib/Pool.js#L450